### PR TITLE
Added Leaderboard toggle button

### PR DESCRIPTION
**Pull request by ACov96**

### CODE REVIEW

**Comment by cabeese** [comment]
When leaderboard is public and admin is logged in, this message is printed to the console:

```
Warning: flattenChildren(...): Encountered two children with the same key, `/leaderboard`. Child keys must be unique; when two children share a key, only the first child will be used.
    in div (created by TopBar)
    in div (created by TopBar)
    in div (created by TopBar)
    in div (created by TopBar)
    in TopBar (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent (created by ReactMeteorDataComponent)
    in ReactMeteorDataComponent (created by App)
    in div (created by App)
    in App (created by RouterContext)
    in RouterContext (created by Router)
    in Router
```

I think the issue is [this line](https://github.com/cabeese/greatpuzzlehunt/blob/1bf4dcf183b012be0a4c1e4c68b0f861c6543c68/client/components/topbar/TopBar.jsx#L158):

```javascript
      <Link key={ item.to } className='item' to={ item.to }>
```

Because two links (one on left side, one on right) have the same key, it gets mad.

File: client/components/topbar/TopBar.jsx [hunk 1]
```diff
@@ -130,7 +138,11 @@ TopBar = class TopBar extends Component {
   }
 
   _renderMenuLinks(links) {
-    return links.map((item) => this._renderMenuLink(item));
+    const { gamestate } = this.props;
+    const linksToRender = links.slice();
```
Comment (cabeese) [inline client/components/topbar/TopBar.jsx:142]: Okay, JavaScript is goofy. This is a new one to me. You can declare an array as `const` but still manipulate the items in it. I understand why, but it still feels misleading...

Comment (ACov96) [reply client/components/topbar/TopBar.jsx:142]: Sorry, this is from my AirBnB ESLint stuff. I can declare it with `let` if that would be better.

**Comment by ACov96** [comment]
I can get rid of the warning if I add the `/admin/leaderboard` route back in. If we go with that approach, then there are two routes that use the same component: `/leaderboard` and `/admin/leaderboard`. It feels like a lot of hoops to jump through to suppress a warning. Not really sure why React cares so much since they are just links...

File: client/components/topbar/TopBar.jsx (hunk 1 above)
Comment (cabeese) [reply client/components/topbar/TopBar.jsx:142]: if it works, it works. I just seems stupid to me that it works. not a big deal though thanks

**Comment by cabeese** [comment]
@ACov96 Not sure exactly what's best. I don't want to screw up React's diffing algorithm somehow... though you're right, it's just a warning. Perhaps I'll make a note about this in Trello and ignore it for now. (aka put it in the icebox and never get around to fixing it)

**Pull request merged**